DESIGN.md

```diff
@@ -0,0 +1,169 @@
+# Overview
+
+Warnet is designed for Bitcoin developers, researchers, and enthusiasts who want to safely experiment with and test ideas related to Bitcoin networks. By leveraging Kubernetes, Warnet enables users to deploy and simulate large-scale Bitcoin networks in a controlled and reproducible environment.
+
+## Table of Contents
+
+1. [Overview](#overview)
+   - [Key Features](#key-features)
+   - [Target Audience](#target-audience)
+   - [Benefits](#benefits)
+
+2. [Philosophy](#philosophy)
+
+3. [Kubernetes and Helm in Warnet](#kubernetes-and-helm-in-warnet)
+   - [Kubernetes](#kubernetes)
+   - [Helm](#helm)
+   - [Why Helm is Preferred in Warnet](#why-helm-is-preferred-in-warnet)
+
+4. [Project Structure](#project-structure)
+   - [Overview of Resources](#overview-of-resources)
+   - [Overview of src/warnet](#overview-of-srcwarnet)
+   - [Overview of Test](#overview-of-test)
+
+5. [Operating in the Network with "Scenarios"/Commanders](#operating-in-the-network-with-scenarioscommanders)
+
+6. [The Resources Configuration Pipeline - An Example](#the-resources-configuration-pipeline---an-example)
+
+## Key Features:
+- Safe simulation of Bitcoin networks
+- Kubernetes-based deployment for scalability and ease of management
+- User interaction capabilities through custom scenarios
+- Infrastructure as Code (IaC) approach for reproducibility
+- Stateless Configuration and Externalized Configuration for flexibility
```

Comment:
I'm not really sure what stateless and externalised configuration is? Is there another way to describe this?